### DISCUSSION TITLE
Require MFA on register?

### DISCUSSION THREAD

**Discussion by danthegoodman1** [ory/kratos#2163] [Q&A] [answered]
Is there a way to enforce that users register MFA for their account within Kratos?

I imagine one's service could check on a user to see whether they have MFA configured in the immediate term like [here](https://www.ory.sh/kratos/docs/two-factor-authentication-2fa-mfa#requesting-second-factor-authentication)

**Comment by aeneasr** [comment, marked as answer]
You can check the user’s session AAL in your middleware and if it is not AAL2 redirect them to the settings page. Your post registration redirect can also point there. Hope this helps :)
